### PR TITLE
feat(dump): create dump endpoint

### LINKED ISSUE #9: add a post endpoint for creating a dump

### PR DESCRIPTION
**Pull request by melancholiai**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✔                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

Closes #9

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ----------------------------------------
 
-A RESTful API for querying osm planet dumps
+A RESTful API for querying and creating osm planet dumps
```
Comment (syncush) [inline README.md:5]: the service does not create osm planet dumps it saves its metadata, I would rephrase that

File: openapi3.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 openapi: 3.0.1
 info:
   title: dump-server
-  description: A restful API for querying osm planet dumps
+  description: A restful API for querying and creating osm planet dumps
```
Comment (syncush) [inline openapi3.yaml:4]: same

File: src/dumpMetadata/controllers/dumpMetadataController.ts [hunk 3]
```diff
@@ -57,4 +59,13 @@ export class DumpMetadataController {
 
     return res.status(httpStatus.OK).json(dumpsMetadata);
   };
+
+  public post: PostDumpMetadataHandler = async (req, res, next) => {
+    try {
+      await this.manager.createDumpMetadata(req.body);
```
Comment (syncush) [inline src/dumpMetadata/controllers/dumpMetadataController.ts:65]: since we do not return the id of the dump metadata record, lets add a log message for indicating a successful creation with the id in it.

File: tests/integration/dumpMetadata/dumpMetadata.spec.ts [hunk 4]
```diff
@@ -19,9 +19,12 @@ import {
   convertFakesToResponses,
   convertFakeToResponse,
   convertToISOTimestamp,
+  createFakeDumpMetadata,
+  createFakeStringByLength,
 } from '../../helpers';
 import { DumpMetadataFilterQueryParams } from '../../../src/dumpMetadata/models/dumpMetadataFilter';
 import { SortFilter } from '../../../src/dumpMetadata/models/dumpMetadataFilter';
+import { BUCKET_NAME_LENGTH_LIMIT, BUCKET_NAME_MIN_LENGTH_LIMIT, DESCRIPTION_LENGTH_LIMIT, NAME_LENGTH_LIMIT } from '../../../src/common/utils/db';
```
Comment (syncush) [inline tests/integration/dumpMetadata/dumpMetadata.spec.ts:27]: I think the `src/common/utils/db` is not great, I think it should be under `src/common/constants`

File: tests/integration/dumpMetadata/dumpMetadata.spec.ts [hunk 5]
```diff
@@ -259,4 +267,108 @@ describe('dumps', function () {
       });
     });
   });
+
+  describe('POST /dumps', function () {
+    describe(`${HAPPY_PATH}`, function () {
+      it('should return 201 status code and create the dump on the db', async function () {
+        const fakeDumpMetada = createFakeDumpMetadata();
+        const { id, ...dumpCreationBody } = fakeDumpMetada;
```
Comment (syncush) [inline tests/integration/dumpMetadata/dumpMetadata.spec.ts:275]: update `@map-colonies/eslint-config` to `v2.2.0` to dispose eslint warning

File: tests/integration/dumpMetadata/dumpMetadata.spec.ts [hunk 6]
```diff
@@ -259,4 +267,108 @@ describe('dumps', function () {
...
+        const response = await requestSender.createDump(app, dumpCreationBody);
+
+        expect(response.status).toBe(httpStatusCodes.CREATED);
+      });
+    });
+
+    describe(`${BAD_PATH}`, function () {
+      it('should return 400 status code if the name is missing', async function () {
+        const fakeDumpMetada = createFakeDumpMetadata();
+        const { id, name, ...dumpCreationBody } = fakeDumpMetada;
```
Comment (syncush) [inline tests/integration/dumpMetadata/dumpMetadata.spec.ts:285]: same

**Review by syncush — CHANGES_REQUESTED** [review]

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (melancholiai) [reply README.md]: done.

File: openapi3.yaml (hunk 2 above)
Comment (melancholiai) [reply openapi3.yaml]: done.

File: src/dumpMetadata/controllers/dumpMetadataController.ts (hunk 3 above)
Comment (melancholiai) [reply src/dumpMetadata/controllers/dumpMetadataController.ts]: done.

File: tests/integration/dumpMetadata/dumpMetadata.spec.ts (hunk 4 above)
Comment (melancholiai) [reply tests/integration/dumpMetadata/dumpMetadata.spec.ts]: done.

File: tests/integration/dumpMetadata/dumpMetadata.spec.ts (hunk 5 above)
Comment (melancholiai) [reply tests/integration/dumpMetadata/dumpMetadata.spec.ts:274]: done.

File: tests/integration/dumpMetadata/dumpMetadata.spec.ts (hunk 6 above)
Comment (melancholiai) [reply tests/integration/dumpMetadata/dumpMetadata.spec.ts:284]: done.

File: tests/helpers/index.ts [hunk 7]
```diff
@@ -25,6 +25,14 @@ export const createFakeDate = (): Date => {
...
 };
 
+export const createFakeStringByLength = (length: number): string => {
+  let str = '';
+  while (str.length < length) {
+    str += faker.lorem.paragraph();
+  }
+  return str.slice(0, length);
+};
+
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply tests/helpers/index.ts]: done, using `faker.random.alpha`

**Review by syncush — APPROVED** [review]
